front_end/dart_tests/dart_support_test.dart

```diff
@@ -1,45 +1,170 @@
-/// Test the dart support functions standalone.
-
-// TODO(alanknight): How do we test things that only run in the context of
-// devtools?
-
-@JS()
 library dart_support_test;
 
 @TestOn('chrome')
 import 'package:test/test.dart';
 import 'package:js/js.dart';
 import 'dart:html';
 import 'dart:async';
+// TODO: Would it be better to pre-compile this and load it from a script tag?
+import 'context_for_evaluation.dart' as context;
+import 'js_interop_for_context_testing.dart';
+import 'test_helpers.dart';
 
-@JS()
-external String $dartExpressionFor(context, String expression);
-
-@JS('eval')
-external jsEval(String expression);
-
-unmodified(String s) {
-  var result = $dartExpressionFor(null, s);
+/// Assert that the dart expression for [s] is [s].
+unmodified(String s) async {
+  var result = await dartExpressionFor(null, s);
   expect(result.split('\n').last.trim(), s);
 }
 
-main() async {
+/// Create a script element for [path] to load it.
+ScriptElement loadScript(String path) {
+  // Note that we have to do this in a separate non-async method
+  // because of https://github.com/dart-lang/sdk/issues/35020
+  var element = new ScriptElement()
+    ..type = 'text/javascript'
+    ..async = false
+    ..src = path;
+  document.body.append(element);
```

Comment:
nit: isn't appending to the header instead of the body better style?

Reply:
Done.